src/classes/Gifts.cls

```diff
@@ -35,26 +35,65 @@
 
 public with sharing class Gifts {
 
-    private Set<String> captureGroupIds;
+    private Set<String> capturableGroupIds;
     private List<DataImport__c> dataImports;
+    @TestVisible
+    private List<DataImport__c> giftsToUpdate = new List<DataImport__c>();
 
     public Gifts(List<DataImport__c> dataImports) {
         this.dataImports = dataImports;
-        setCaptureGroupIds(dataImports);
+        setCapturableGroupIds(dataImports);
     }
 
-    private void setCaptureGroupIds(List<DataImport__c> dataImports) {
-        captureGroupIds = new Set<String>();
-        for (DataImport__c dataImport : dataImports) {
-            captureGroupIds.add(dataImport.Payment_Elevate_Batch_ID__c);
+    public List<DataImport__c> authorizedGifts() {
+        List<DataImport__c> giftsToReturn = new List<DataImport__c>();
+        for (DataImport__c gift : dataImports) {
+            if (gift.Payment_Status__c == ElevateTransactionStatus.AUTHORIZED.name()) {
+                giftsToReturn.add(gift);
+            }
+        }
+        return giftsToReturn;
+    }
+
+    public Set<String> capturableGroupIds() {
+        return capturableGroupIds;
+    }
+
+    public void authorizedToPaymentProcessingStatus() {
+        for (DataImport__c gift : authorizedGifts()) {
+            gift.Payment_Status__c = ElevateTransactionStatus.PAYMENTPROCESSING.name();
+            giftsToUpdate.add(gift);
+        }
+    }
+
+    public void failGiftsByCaptureIdWithReason(Map<String, String> failReasonByCaptureGroupId) {
+        for (String captureGroupId : failReasonByCaptureGroupId.keySet()) {
+            DataImport__c gift = findByCaptureGroupId(captureGroupId);
```

Comment:
In future state, wouldn't there be several DIs/gifts that match a single capture group Id? Are we locking ourselves into the 1-to-1 relationship here?

Reply:
You're absolutely correct! Updated to take that into account. I tried to avoid doing an inner loop, but I circled back to it being ok in this case. Split the other loop out into a private method for readability. Let me know what you think. Additionally added test coverage for this class as well